protobuf/protoc-gen-gograin/template.go

```diff
@@ -70,30 +71,29 @@ func (g *{{ $service.Name }}Grain) {{ $method.Name }}WithOpts(r *{{ $method.Inpu
 			}
 			switch msg := response.(type) {
 			case *cluster.GrainResponse:
-				result := &{{ $method.Output.Name }}{}
 				err = proto.Unmarshal(msg.MessageData, result)
 				if err != nil {
-					return nil, err
+					return result, err
 				}
 				return result, nil
 			case *cluster.GrainErrorResponse:
-				return nil, errors.New(msg.Err)
+				return result, errors.New(msg.Err)
 			default:
-				return nil, errors.New("unknown response")
+				return result, errors.New("unknown response")
 			}
 		}
 	
 	var res *{{ $method.Output.Name }}
 	var err error
 	for i := 0; i < opts.RetryCount; i++ {
 		res, err = fun()
-		if err == nil {
+		if err == nil || err.Error() != "future: timeout" {
```

Comment:
@otherview Thank you for the PR and really sorry for late reply! = )
I'm not so sure on this one. If there is an error, even if it's not the future: timeout one, we should still return nil instead of an empty object?

Reply:
Good spot! Oversight when moving this from the test code.

That should be returning the err :) 👍

Reply:
@otherview Thank you for revision! = ) Just another question, is there any reason to move `result := &{{ $method.Output.Name }}{}` to the top, and return the empty object instead of `nil` when receiving error? Thx! = )

Reply:
Good point.

I think my initial reasoning at first was to mimic the behaviour of protobuf where you always return something.
Now.. I don't think it adds much to be honest. 
Happy to remove if you think the same 👍

Reply:
@otherview Sure! Sorry I got so many requests, but I think `return nil` should suit the situation better here... Thank you very much!! = )

Reply:
No problem @PotterDai and thanks for your input man :)